src/Eloquent/HasRecursiveRelationships.php

```diff
@@ -2,11 +2,41 @@
 
 namespace Staudenmeir\LaravelAdjacencyList\Eloquent;
 
+use Illuminate\Database\Eloquent\Model;
 use Staudenmeir\LaravelAdjacencyList\Eloquent\Traits\HasAdjacencyList;
 use Staudenmeir\LaravelCte\Eloquent\QueriesExpressions;
 
 trait HasRecursiveRelationships
 {
     use HasAdjacencyList;
     use QueriesExpressions;
+
+    public function isParentOf(Model $model)
+    {
+        if (!$this->relationLoaded('children')) {
```

Comment:
The `if {}` here (and in the other methods) is not necessary. `$this->children` automatically loads the relationship if necessary.